test/lib/ansible_test/_internal/sanity/import.py

```diff
@@ -92,91 +103,112 @@ def test(self, args, targets, python_version):
+                run_command(args, generate_pip_install(virtualenv_pip, '', packages=['setuptools']), env=env, capture=capture_pip)
+                run_command(args, generate_pip_install(virtualenv_pip, '', packages=['coverage']), env=env, capture=capture_pip)
+
+            try:
+                # In some environments pkg_resources is installed as a separate pip package which needs to be removed.
+                # For example, using Python 3.8 on Ubuntu 18.04 a virtualenv is created with only pip and setuptools.
+                # However, a venv is created with an additional pkg-resources package which is independent of setuptools.
+                # Making sure pkg-resources is removed preserves the import test consistency between venv and virtualenv.
+                # Additionally, in the above example, the pyparsing package vendored with pkg-resources is out-of-date and generates deprecation warnings.
+                # Thus it is important to remove pkg-resources to prevent system installed packages from generating deprecation warnings.
+                run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'pkg-resources'], env=env, capture=capture_pip)
+            except SubprocessError:
+                pass
+
+            run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'setuptools'], env=env, capture=capture_pip)
+            run_command(args, virtualenv_pip + ['uninstall', '--disable-pip-version-check', '-y', 'pip'], env=env, capture=capture_pip)
+
+            display.info(import_type + ': ' + data, verbosity=4)
+
+            cmd = ['importer.py', '--type', import_type]
+
+            this_results = []
+            try:
+                with coverage_context(args):
+                    stdout, stderr = intercept_command(args, cmd, self.name, env, capture=True, data=data, python_version=python_version,
+                                                       virtualenv=virtualenv_python)
+
+                if stdout or stderr:
+                    raise SubprocessError(cmd, stdout=stdout, stderr=stderr)
+            except SubprocessError as ex:
+                if ex.status != 10 or ex.stderr or not ex.stdout:
+                    raise
+
+                pattern = r'^(?P<path>[^:]*):(?P<line>[0-9]+):(?P<column>[0-9]+): (?P<message>.*)$'
+
+                this_results = parse_to_list_of_dict(pattern, ex.stdout)
+
+                relative_temp_root = os.path.relpath(temp_root, data_context().content.root) + os.path.sep
+
+                this_results = [SanityMessage(
+                    message=r['message'],
+                    path=os.path.relpath(r['path'], relative_temp_root) if r['path'].startswith(relative_temp_root) else r['path'],
+                    line=int(r['line']),
+                    column=int(r['column']),
+                ) for r in this_results]
+
+            results.extend(settings.process_errors(this_results, paths))
```

Comment:
Move this outside the loop:

```suggestion
        results = settings.process_errors(messages, paths)
```